suite-native/atoms/src/Sheet/BottomSheetFlashList.tsx

```diff
@@ -0,0 +1,146 @@
+import { useEffect, useRef, ReactNode } from 'react';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import Animated from 'react-native-reanimated';
+import { PanGestureHandler } from 'react-native-gesture-handler';
+import { GestureResponderEvent, Pressable } from 'react-native';
+
+import { FlashList, FlashListProps } from '@shopify/flash-list';
+
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+import { Box } from '../Box';
+import { BottomSheetContainer } from './BottomSheetContainer';
+import { useBottomSheetAnimation } from './useBottomSheetAnimation';
+import { BottomSheetHeader } from './BottomSheetHeader';
+
+export type BottomSheetFlashListProps<TItem> = {
+    isVisible: boolean;
+    isCloseDisplayed?: boolean;
+    onClose: (isVisible: boolean) => void;
+    title?: ReactNode;
+    subtitle?: ReactNode;
+    ExtraProvider?: React.ComponentType;
+    estimatedListHeight?: number;
+} & FlashListProps<TItem>;
+
+const DEFAULT_INSET_BOTTOM = 50;
+
+const sheetWrapperStyle = prepareNativeStyle(utils => ({
+    backgroundColor: utils.colors.backgroundSurfaceElevation0,
+    borderTopLeftRadius: utils.borders.radii.large,
+    borderTopRightRadius: utils.borders.radii.large,
+    maxHeight: '80%',
+}));
+
+const sheetContentContainerStyle = prepareNativeStyle<{
+    insetBottom: number;
+}>((utils, { insetBottom }) => ({
+    paddingBottom: Math.max(insetBottom, utils.spacings.sp16),
+    paddingHorizontal: utils.spacings.sp16,
+}));
+
+const sheetWithOverlayStyle = prepareNativeStyle(_ => ({
+    flex: 1,
+    justifyContent: 'flex-end',
+}));
+
+export const BottomSheetFlashList = <TItem,>({
+    isVisible,
+    isCloseDisplayed = true,
+    onClose,
+    title,
+    subtitle,
+    ExtraProvider,
+    estimatedListHeight = 0,
+    ...flashListProps
+}: BottomSheetFlashListProps<TItem>) => {
+    const { applyStyle } = useNativeStyles();
+    const insets = useSafeAreaInsets();
+    const {
+        animatedSheetWithOverlayStyle,
+        animatedSheetWrapperStyle,
+        closeSheetAnimated,
+        openSheetAnimated,
+        panGestureEvent,
+        scrollEvent,
+    } = useBottomSheetAnimation({
+        onClose,
+        isVisible,
+    });
+    const panGestureRef = useRef(null);
+    const scrollViewRef = useRef(null);
+
+    useEffect(() => {
+        if (isVisible) {
+            openSheetAnimated();
+        }
+    }, [isVisible, openSheetAnimated]);
+
+    const handleCloseSheet = () => {
+        closeSheetAnimated();
+    };
+
+    const handlePressOutside = (event: GestureResponderEvent) => {
+        if (event.target === event.currentTarget) closeSheetAnimated();
+    };
+
+    const insetBottom = Math.max(insets.bottom, DEFAULT_INSET_BOTTOM);
+
+    return (
+        <BottomSheetContainer
+            isVisible={isVisible}
+            onClose={handleCloseSheet}
```

Comment:
What about using `closeSheetAnimated` here directly? 🤔 
```suggestion
            onClose={closeSheetAnimated}
```

Reply:
Make sense